src/__tests__/ParseObject-test.js

```diff
@@ -2024,6 +2024,142 @@ describe('ParseObject', () => {
     }
   });
 
+  it('should fail save with transaction and batchSize option', async () => {
+    const obj1 = new ParseObject('TestObject');
+    const obj2 = new ParseObject('TestObject');
+
+    try {
+      await ParseObject.saveAll([obj1, obj2], { transaction: true, batchSize: 20 });
+      expect(true).toBe(false);
+    } catch (e) {
+      expect(e.message).toBe(
+        'You cannot use both transaction and batchSize options simultaneously.'
+      );
+    }
+  });
+
+  it('should fail destroy with transaction and batchSize option', async () => {
+    const obj1 = new ParseObject('TestObject');
+    const obj2 = new ParseObject('TestObject');
+
+    try {
+      await ParseObject.destroyAll([obj1, obj2], { transaction: true, batchSize: 20 });
+      expect(true).toBe(false);
+    } catch (e) {
+      expect(e.message).toBe(
+        'You cannot use both transaction and batchSize options simultaneously.'
+      );
+    }
+  });
+
+  it('should fail save batch with unserializable attribute and transaction option', async () => {
+    const obj1 = new ParseObject('TestObject');
+    const obj2 = new ParseObject('TestObject');
+    obj1.set('relatedObject', obj2);
+
+    try {
+      await ParseObject.saveAll([obj1, obj2], { transaction: true });
+      expect(true).toBe(false);
```

Comment:
Could you rewrite this and similar lines with `await expectAsync().toBeRejectedWith(...)`?

Reply:
I initially intended to write it as you suggested with await expectAsync().toBeRejectedWith(...). However, I noticed that we have many instances of this type of checking throughout the tests in the repo. To maintain consistency, I followed the existing pattern.

That said, we could consider updating all of them in one go in a separate PR, as there are about 25 similar instances in other files. For now, I'll go ahead and commit the changes to fix these specific cases, and perhaps we can address the rest incrementally later.

Reply:
I updated these tests, can you check them out again?